src/main/java/org/myrobotlab/service/Amt203Encoder.java

```diff
@@ -25,16 +25,17 @@
  * @author kwatters
  *
  */
-public class Amt203Encoder extends AbstractEncoder implements EncoderControl {
+public class Amt203Encoder extends AbstractPinEncoder implements EncoderControl {

```

Comment:
As mentioned in the shoutbox.  Amt203Encoder isn't really pin based.. it's SPI based.  I don't think this is a good name for the base class.  The base class should just be AbstractEncoder.

Reply:
When I met it - it required a pin .. there were references to either - encoder.setPin(3); or encoder.pin = 3; ... 
I guess we both agree this shouldn't really be in the interface as is part of configuration. Its possible an encoder could have x pins to properly configure or no pins (as the case with TimeEncoder) ..  so I would think you would agree it shouldn't be in AbstractEncoder - hence the move